### PR TITLE
Don't set null in set_node newProperties when using unsetNodes

### PR DESCRIPTION
**Pull request by TheSpyder**
**Description**
Having `null` values in `set_node` operations is unnecessary, thanks to #4078, and adds complexity to collaboration transforms.

**Issue**
Follow up to #4078

**Context**
After #4078 was submitted @ianstormtaylor suggested omitting removed keys, rather than setting them to `null`, which was a much cleaner solution. This tidies up the last place that null values might be set.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [x] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

**Review by dylans — APPROVED** [review]
looks reasonable to me.

**Comment by TheSpyder** [comment]
The CI failure seems to be an inconsistent test, it was [recently bumped to a 10s timeout](https://github.com/ianstormtaylor/slate/pull/4423) but it seems to not be enough sometimes. I suspect it's actually failing for some unexpected reason if 10 seconds isn't enough.

**Comment by TheSpyder** [comment]
Given that the idea was pre-approved by Ian in #4078, I'll merge it now - easy to revert if other maintainers spot problems later.

**Comment by dylans** [comment]
> Given that the idea was pre-approved by Ian in #4078, I'll merge it now - easy to revert if other maintainers spot problems later.

The failing integration test is not the result of your change, so feel free to squash & merge.

**Comment by TheSpyder** [comment]
Holding for a potential patch release before this minor bump

**Pull request merged**
